### DISCUSSION TITLE
Slow add-on updates

### DISCUSSION THREAD

**Discussion by morhimi** [hassio-addons/addon-zwavejs2mqtt#465] [General]
Hi,
any reason why there is no auto update workflow that creates a new addon when there is a new version of the zwave-js-ui ?
just wondering since I see that project has a much higher release cadence while updates reach HA very slowly
thanks...

**Comment by morhimi** [comment]
btw, I see there is a PR open now
https://github.com/hassio-addons/addon-zwave-js-ui/pull/463
just wondering if this can be automated somehow or if the community can help here?

**Reply by frenck** [reply]
> Hi,
any reason why there is no auto update workflow that creates a new addon when there is a new version of the zwave-js-ui ?

Because it needs testing.

> btw, I see there is a PR open now

I processed parts of that PR around an hour ago. However, to prove the above point, as you see in the PR  you've linked... it isn't tested... the build is failing. That is why additional validation in done in between. I won't be shipping anything that is untested. Nor can I ship anything that isn't building...

That said, if you don't like the speed, feel free to build and manage your own.

../Frenck

**Reply by morhimi** [reply]
@frenck I probably should have explained myself better, if it sounded too harsh, my apologies.
for clarification, I saw that PR *after* I wrote the initial message,  and I gave it as an example of something good, and that's what I referred to when asking "if the community can help".

so for example, would it help if someone will open a PR as soon as there is a new ZW JS UI version available? (or maybe it will jsut burden the maintainers) 
assuming build doesn't fail, is there a way for the community to assist in testing?
does it make sense to create a beta channel for this addon?

I'm fully in favor of testing (and the obvious of not shipping DOAs), I'm just addressing this from a place where maybe more people could help.

I of course appreciate all the hard work you do on this addon and HA in general.
--Mor

**Reply by frenck** [reply]
> would it help if someone will open a PR as soon as there is a new ZW JS UI version available? 

Sure, it is open source, feel free to do so..

> assuming build doesn't fail, is there a way for the community to assist in testing?

Every PR should be tested before creating one. The PR now, is an example of a drive-by PR, which is really not helping anybody.

I will, nevertheless, review and test every PR.

> does it make sense to create a beta channel for this addon?

It already has, as a matter of fact, all my add-ons have 3 channels, edge, beta, and stable. If you invested time into looking into what is there instead of raising all these suggestions, you'd have saved me time in this discussion as well.

> I'm just addressing this from a place where maybe more people could help.

Thanks! 👍 

../Frenck
